lib/foldersummary/folder_summary.go

```diff
@@ -4,24 +4,38 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // You can obtain one at https://mozilla.org/MPL/2.0/.
 
-package main
+package foldersummary
 
 import (
+	"fmt"
+	"strings"
 	"time"
 
+	"github.com/syncthing/syncthing/lib/config"
 	"github.com/syncthing/syncthing/lib/events"
+	"github.com/syncthing/syncthing/lib/model"
 	"github.com/syncthing/syncthing/lib/protocol"
 	"github.com/syncthing/syncthing/lib/sync"
+	"github.com/syncthing/syncthing/lib/ur"
 	"github.com/thejerf/suture"
 )
 
+const DefaultEventTimeout = time.Minute
+
+type Service interface {
+	suture.Service
+	FolderSummary(folder string) (map[string]interface{}, error)
+	GotEventRequest()
+}
+
 // The folderSummaryService adds summary information events (FolderSummary and
 // FolderCompletion) into the event stream at certain intervals.
-type folderSummaryService struct {
+type service struct {
 	*suture.Supervisor
 
-	cfg       configIntf
-	model     modelIntf
+	cfg       ur.ConfigIntf
+	model     ur.ModelIntf
```

Comment:
Having to reach out into UR is GRIMMMMMM

Reply:
mmm this seems like too small packages perhaps? But I agree, this dependency edge is ugly.

Reply:
That's got a point, should `lib/foldersummary` go to `lib/ur`? It also a report of sorts. Will just mean that `ur.Service` will have to be come `ur.UsageReportingService` to prevent ambiguities with `foldersummary.Service` which then becomes `ur.FolderSummaryService`.

Reply:
Or `lib/model`? I know moving stuff to `lib/model` is the opposite way of what we usually do, but thematically it seems right to have it with all the rest of the folder code (moving folders to lib/folder might also be viable at some point).